weatherApp/App/UI/Views/MainInformationView.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  MainInformationView.swift
+//  weatherApp
+//
+//  Created by Dario Nikolic on 25/08/2020.
+//  Copyright © 2020 Dario Nikolic. All rights reserved.
+//
+
+import UIKit
+import PureLayout
+import Kingfisher
+
+class MainInformationView: UIView {
+    
+    private let locationNameLabel = UILabel()
+    private let weatherDescriptionLabel = UILabel()
+    private let currentTempLabel = UILabel()
+    private let minTempLabel = UILabel()
+    private let maxTempLabel = UILabel()
+    private let weatherIcon = UIImageView()
+    private let temperatureStackView = UIStackView()
+    private let lineView = UIView()
+    
+    override init(frame: CGRect) {
+         super.init(frame: frame)
+         commonInit()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+         super.init(coder: aDecoder)
+         commonInit()
+    }
+    
+    func set(currentWeather: CurrentWeatherViewModel) {
+        locationNameLabel.text = currentWeather.name
+        weatherDescriptionLabel.text = currentWeather.weatherDescription
+        currentTempLabel.text = String(format: LocalizedStrings.degreeValueFormat, currentWeather.temperature)
+        minTempLabel.text = String(format: LocalizedStrings.temperatureValueFormat, currentWeather.minTemperature)
+        maxTempLabel.text = String(format: LocalizedStrings.temperatureValueFormat, currentWeather.maxTemperature)
+        
+        let urlString = currentWeather.weatherIconUrlString
+        if let url = URL(string: urlString) {
+            weatherIcon.kf.setImage(with: url)
+        }
+    }
+    
+    private func commonInit() {
+        setupSubviews()
+        styleElements()
+        setupLayout()
+        
+        backgroundColor = .clear
+    }
+    
+    //MARK: - Styling Elements
+    
+    private func styleElements() {
```

Comment:
You can move style and auto-layout in a separate extension which can look like:
```
extension MainInformationview {
   func buildViews() {
        createViews()
        styleViews()
        defineLayoutForViews()
    }

   func createViews() {
      weatherIcon = UIImageView()
      addSubview(weatherIcon)
    }

   func styleViews() {
      // Move all private style functions in this extension and call them from here
   }

   func defineLayoutForViews() {
      weatherIcon.autoPinEdge(.top, to: .bottom, of: locationNameLabel, withOffset: 10)
      weatherIcon.autoPinEdge(.leading, to: .leading, of: self, withOffset: 20)
   }
}
```

After you create an extension you can call buildViews() instead of commonInit().

Reply:
Should an extension be in a separate file or in the same file?